msg_test.go

```diff
@@ -324,3 +324,89 @@ func TestLenDynamicA(t *testing.T) {
 		}
 	}
 }
+
+func TestUnPackForErrBuf(t *testing.T) {
```

Comment:
Sure, I have added a shorter test case.

Reply:
thanks.
But you don't need to set up a localserver, only making a message with a bunch of RRs, adjust the counters in the raw byte slice and then call unpack (func (dns *Msg) Unpack(msg []byte)) and check the err
And another PR we're talking about ErrIllegal or something, we don't have that yet, but might make sense to use that here at some point

Reply:
Sure, I have followed your suggestion. 

In case we decide to use ErrIllegal or something later on, I will have to make a PR in coredns repo to handle it. For ErrBuf, there is no change in coredns needed - Reference PR - https://github.com/coredns/coredns/pull/6277